src/main/webapp/app/shared/list/FormListField.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import styles from './styles.module.scss';
+import { Row, Col, Button } from 'react-bootstrap';
+import { AvGroup, AvInput } from 'availity-reactstrap-validation';
+import autobind from 'autobind-decorator';
+import { action, observable } from 'mobx';
+
+type IListBoxProps = {
+  list: string[];
+  onDelete: (item: string) => void;
+};
+
+@observer
+class ListBox extends React.Component<IListBoxProps> {
+  render() {
+    return (
+      <div
+        style={{
+          border: '1px solid #ced4da',
+          borderRadius: '0.25rem',
+          padding: '0.375rem 0.75rem',
+          overflowY: 'scroll',
+          minHeight: '100px',
+          maxHeight: '100px',
+          marginTop: '28px',
+        }}
+      >
+        <div className={styles.main}>
+          {this.props.list.map(item => (
+            <div key={item} className={styles.content_container}>
+              <span
+                className={styles.delete}
+                onClick={() => this.props.onDelete(item)}
+              >
+                <i className="fa fa-times-circle"></i>
+              </span>
+              <span style={{ margin: '5px 10px 5px 5px' }}>{item}</span>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+type IFormListFieldProps = {
+  list: string[];
+  addItem: (item: string) => void;
+  deleteItem: (item: string) => void;
+  labelText: string;
+  placeholder?: string;
+  boldLabeL?: boolean;
+};
+
+@observer
+export class FormListField extends React.Component<IFormListFieldProps> {
+  @observable isInputValid = false;
+  @observable inputText = '';
+  @observable errorMessage = '';
+
+  @autobind
+  listLengthValidation(value: any, context: any, input: any, cb: any) {
+    if (this.props.list.length > 0) {
+      cb(true);
+    } else {
+      cb(false);
+      this.errorMessage = 'List cannot be empty!';
+    }
+  }
+
+  @action.bound
+  domainValidation() {
+    if (this.inputText.length === 0) {
+      this.errorMessage = '';
+      return false;
+    } else if (
+      this.props.list.includes(this.inputText) ||
+      !this.inputText.includes('.')
+    ) {
+      this.errorMessage = 'Domain is not valid';
+      return false;
+    }
+    this.errorMessage = '';
+    return true;
+  }
+
+  @action.bound
+  onEnterKey(event: any) {
+    if (event.which === 13 && this.isInputValid) {
+      this.props.addItem(this.inputText);
+      this.inputText = '';
+      this.isInputValid = this.domainValidation();
+    }
+  }
+
+  render() {
+    return (
+      <AvGroup>
+        <Row>
+          <Col>
+            <label
+              className={this.props.boldLabeL ? 'font-weight-bold' : ''}
+              htmlFor="listItemInput"
+            >
+              {this.props.labelText}
+            </label>
+            <div className="input-group">
+              <AvInput
+                id="listItemInput"
```

Comment:
I'm sure the id will be duplicated if multiple FormListField is required. If we cannot avoid using the id, maybe we should require a `id` attr on the component?

Reply:
Good point. I used the `id` attribute on `AvInput`, so that the label is bound to it.